index.js

```diff
@@ -1,79 +1,157 @@
+console.log(fullname4); // returns "Lady Alice Johnson"
+//Event application
+
+function getEventWeekday(daysFromToday) {
+  // Array of weekdays
+  const weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+
+  // Get today's date and weekday
+  const today = new Date();
+  const todayWeekday = today.getDay(); // 0 (Sunday) to 6 (Saturday)
+
+  // Calculate the future weekday
+  const eventWeekday = (todayWeekday + daysFromToday) % 7;
+
+  // Return the name of the weekday
+  return weekdays[eventWeekday];
+}
+
+// Example usage:
+console.log(getEventWeekday(9)); // Logs out the weekday 9 days from today
+console.log(getEventWeekday(2)); // Logs out the weekday 2 days from today
+
+
 
-// Step 3: Create the startup name
-const startupName = firstWords[randomIndex1] + " " + secondWords[randomIndex2];
+//Weather wear
+;
+function youCreateThisFunctionName(temperature){
+   if(temperature <= 0){
+    return "Sweater and hat and warm jacket";
+   }else if(temperature >0 && temperature <= 15){
+    return "Sweater and jacket"
+   }else if(temperature >15 && temperature <= 20){
+    return "T-shirt and jacket"
+   } else if(temperature >20 && temperature <=25){
+    return  "T-shirt and pants"
+   }else{return "T-shirts and shorts"}
 
-// Step 4: Log the startup name and its character count
-console.log("The startup: \"" + startupName + "\" contains " + startupName.length + " characters.");
+}
+  const clothesToWear = youCreateThisFunctionName(15);
+console.log(clothesToWear);
+
+
+
+//Student manager
+let class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (studentName === "") {
+        console.log("Cannot add an empty string as a student name.");
+        return;
+    }
+
+    if (class07Students.includes(studentName)) {
+        console.log(`Student ${studentName} is already in the class`);
+        return;
+    }
+
+    if (class07Students.length >= 6 && studentName !== "Queen") {
+        console.log("Cannot add more students to class 07");
+        return;
+    }
+
+    class07Students.push(studentName);
+    console.log(class07Students);
+}
+
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+
+// Example usage:
+addStudentToClass("Benjamin"); // Adds Benjamin to the class
+addStudentToClass("Alice");    // Adds Alice to the class
+addStudentToClass("");         // Logs: Cannot add an empty string as a student name.
+addStudentToClass("Alice");    // Logs: Student Alice is already in the class
+addStudentToClass("John");
+addStudentToClass("Jane");
+addStudentToClass("Mariya");
+addStudentToClass("Jerry");    // Logs: Cannot add more students to class 07
+addStudentToClass("Queen");    // Always adds Queen even if the class is full
+
+
+ console.log(getNumberOfStudents());
   
+//Candy helper optional
+
+let boughtCandyPrices=[]
+
+function addCandy(candyType,weight){
+boughtCandyPrices.push(candyType,weight)
+}
```

Comment:
In this exercise, to calculate the total money spend we want the `boughtCandyPrices` array contain the amounts spent after each purchase. 

Here is an example: let's say I am buying **100g** of **Sweet candy**. The price per gram is **0.5**, so the total I spend is **50**.

```js
let boughtCandyPrices=[]

addCandy("sweet", 100) // the array should become [50]
addCandy("chocolate", 100) // now, it's [50, 70], 0.7 per gram for chocolate candy
```

When you call `push()` function with two arguments it will actually add two element to the array instead, one of which is going to be a string. Try playing around with the `push()` function in the console.

```js
// candyType is "sweet" and weight is 100
boughtCandyPrices.push(candyType,weight);

console.log(boughtCandyPrices); // ["sweet",100] which is not what we want
```

I think what can be helpful here is to write a function that will return a price per gram.

 ```suggestion
function addCandy(candyType,weight){
  let pricePerGram = pricePerGramOfCandy(candyType);
  boughtCandyPrices.push(pricePerGram * weight)
}
```

Can you write this function? Let me know if you need help with that.